svgast/units.py

```diff
@@ -22,6 +22,66 @@ def __str__(self):
             str_number(self.px / self._px_per_unit),
             self.unit)
 
+    def __abs__(self):
+        return self.px
+
+    def __add__(self, other):
+        if isinstance(other, Length):
+            px = self.px + other.px
+        else:
+            px = self.px + other
+        l = user(px)
+        if type(self) is type(other):
+            return type(self)(l)
+        else:
+            return l
+
+    def __radd__(self, other):
+        assert not isinstance(other, Length)
+        return user(other + self.px)
+
+    def __sub__(self, other):
+        if isinstance(other, Length):
+            px = self.px - other.px
+        else:
+            px = self.px - other
+        l = user(px)
+        if type(self) is type(other):
+            return type(self)(l)
+        else:
+            return l
+
+    def __rsub__(self, other):
+        assert not isinstance(other, Length)
+        return user(other - self.px)
+
+    def __mul__(self, other):
+        if isinstance(other, Complex):
+            return type(self)(user(self.px * other))
+        else:
+            raise TypeError()
+
+    def __rmul__(self, other):
+        if isinstance(other, Complex):
+            return type(self)(user(other * self.px))
+        else:
+            raise TypeError()
+
+    def __truediv__(self, other):
+        if isinstance(other, Length):
```

Comment:
This isn't quite the opposite of multiplication?

Reply:
Yeah, it doesn't allow you to flip the operands like multiplication does. Dividing a scalar by a united length seemed weird. I was running off the idea that units like `mm^-1` and `cm^2` were to be avoided, but anything that gave you a regular "1D" unit was fine.

Reply:
I just meant it looked different in type terms, although I think duck typing might make it equivalent.